File: test/formats/gfx/sotes_converter_test.rb
require 'lib/formats/gfx/sotes_converter'
require 'lib/virtual_file'
require 'test/test_helper'

# Unit tests for SotesConverter
class SotesConverterTest < Test::Unit::TestCase
  def test_decoding_palette
    input_data = TestHelper.get_test_file('sotes/#1726')
    expected_data = TestHelper.get_test_file('sotes/#1726-out.png')

    file = VirtualFile.new(nil, input_data)
    SotesConverter.decode!(file, {})
    actual_data = file.data

    TestHelper.compare_image(expected_data, actual_data)
  end

  def test_decoding_non_palette
    input_data = TestHelper.get_test_file('sotes/#1410')
    expected_data = TestHelper.get_test_file('sotes/#1410-out.png')

    file = VirtualFile.new(nil, input_data)
    SotesConverter.decode!(file, {})
    actual_data = file.data

    TestHelper.compare_image(expected_data, actual_data)
  end
end

File: tests/formats/gfx/sotes_converter_test.cc
#include "formats/gfx/sotes_converter.h"
#include "test_support/converter_support.h"

void test_sotes_decoding(
    const std::string &input_path, const std::string expected_path)
{
    SotesConverter converter;
    assert_decoded_image(converter, input_path, expected_path);
}

int main(void)
{
    test_sotes_decoding(
        "tests/test_files/gfx/sotes/#1410",
        "tests/test_files/gfx/sotes/#1410-out.png");
    test_sotes_decoding(
        "tests/test_files/gfx/sotes/#1726",
        "tests/test_files/gfx/sotes/#1726-out.png");
    return 0;
}
